pennylane_lightning/lightning_kokkos/_measurements.py

```diff
@@ -0,0 +1,465 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Class implementation for state vector measurements.
+"""
+
+# pylint: disable=import-error, no-name-in-module, ungrouped-imports
+try:
+    from pennylane_lightning.lightning_kokkos_ops import MeasurementsC64, MeasurementsC128
+except ImportError:
+    pass
+
+from functools import reduce
+from typing import Callable, List, Union
+
+import numpy as np
+import pennylane as qml
+from pennylane.devices.qubit.sampling import _group_measurements
+from pennylane.measurements import (
+    ClassicalShadowMP,
+    CountsMP,
+    ExpectationMP,
+    MeasurementProcess,
+    ProbabilityMP,
+    SampleMeasurement,
+    ShadowExpvalMP,
+    Shots,
+    StateMeasurement,
+    VarianceMP,
+)
+from pennylane.ops import Hamiltonian, SparseHamiltonian, Sum
+from pennylane.tape import QuantumScript
+from pennylane.typing import Result, TensorLike
+from pennylane.wires import Wires
+
+from pennylane_lightning.core._serialize import QuantumScriptSerializer
+
+
+class LightningKokkosMeasurements:
+    """Lightning Measurements class
+
+    Measures the state provided by the LightningStateVector class.
+
+    Args:
+        qubit_state(LightningStateVector): Lightning state-vector class containing the state vector to be measured.
+    """
+
+    def __init__(
+        self,
+        kokkos_state,
+    ) -> None:
+        self._qubit_state = kokkos_state
+        self._dtype = kokkos_state.dtype
+        self._measurement_lightning = self._measurement_dtype()(kokkos_state.state_vector)
+
+    @property
+    def qubit_state(self):
+        """Returns a handle to the LightningStateVector class."""
```

Comment:
Is it more accurate if we change LightningStateVector class to `LightningStateVector` object?